to.etc.domui/src/main/java/to/etc/domui/component/layout/TabPanelBase.java

```diff
@@ -60,6 +60,13 @@ public interface ITabSelected {
 		void onTabSelected(TabPanelBase tabPanel, int oldTabIndex, int newTabIndex) throws Exception;
 	}
 
+	/**
+	 * Represents check if tab can be selected.
+	 */
+	public interface ITabSelectable {
+		boolean isTabSelectable(TabPanelBase tabPanel, int oldTabIndex, int newTabIndex) throws Exception;
```

Comment:
Use TabInstance instead of int to indicate the from/to change, do not use indices. I also think that it is better not to add the tabpanel itself; that type of parameter is almost never used and is very easy to provide in the listener itself.